packages/fetchai/skills/oracle_aggregation/behaviours.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the behaviours for the oracle aggregation skill."""
+
+from typing import cast
+
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.protocols.oef_search.message import OefSearchMessage
+from packages.fetchai.skills.oracle_aggregation.dialogues import OefSearchDialogues
+from packages.fetchai.skills.oracle_aggregation.strategy import GenericStrategy
+
+
+DEFAULT_SEARCH_INTERVAL = 5.0
+
+
+class GenericSearchBehaviour(TickerBehaviour):
+    """This class implements a search behaviour."""
+
+    def __init__(self, **kwargs):
+        """Initialize the search behaviour."""
+        search_interval = cast(
+            float, kwargs.pop("search_interval", DEFAULT_SEARCH_INTERVAL)
+        )
+        super().__init__(tick_interval=search_interval, **kwargs)
+
+    def setup(self) -> None:
+        """Implement the setup for the behaviour."""
+        strategy = cast(GenericStrategy, self.context.strategy)
+        self._register_agent()
+        self._register_service_personality_classification()
+        strategy.make_observation()
```

Comment:
Unless it is actual setup, I would avoid doing it here. You'll probably want a separate behaviour which lets you configure this particular aspect independently of the search frequency.